### PR TITLE
prepare docs for 5.1

### PR DESCRIPTION
**Pull request by minrk**
cc @willingc

I do want to merge the example notebooks into the documentation like we now do on nbconvert,
but not for this release.

### CODE REVIEW

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -3,6 +3,47 @@
 Changes in IPython Parallel
 ===========================
 
+
+5.1
+---
+
```
Comment (willingc) [inline docs/source/changelog.rst]: You might consider adding an explicit subtitle for "New features".

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -3,6 +3,47 @@
...
+IPython Parallel 5.1 adds integration with other parallel computing tools,
+such as `dask.distributed <https://distributed.readthedocs.io>`_ and `joblib <https://pythonhosted.org/joblib>`__.
+
+To turn an IPython cluster into a dask.distributed cluster,
```
Comment (willingc) [inline docs/source/changelog.rst]: Perhaps subtitle "dask.distributed"

File: docs/source/changelog.rst [hunk 3]
```diff
@@ -3,6 +3,47 @@
...
+call :meth:`~.Client.become_distributed`::
+
+    executor = client.become_distributed(ncores=1)
+
+which returns a :class:`distributed:~.Executor` instance.
+
+To register IPython Parallel as the backend for joblib::
```
Comment (willingc) [inline docs/source/changelog.rst]: Perhaps subtitle "joblib"

File: docs/source/changelog.rst [hunk 4]
```diff
@@ -3,6 +3,47 @@
...
+    jupyter nbextension install --py ipyparallel
+    jupyter nbextension enable --py ipyparallel
+
+though you can still use the more convenient single-call::
+
+    ipcluster nbextension enable
+
+which does all three steps above.
+
+`Slurm <https://computing.llnl.gov/tutorials/linux_clusters>`_ support is added to ipcluster.
```
Comment (willingc) [inline docs/source/changelog.rst]: Perhaps subtitle "Slurm support"

**Comment by willingc** [comment]
@minrk Overall, this looks good. Whether by subtitles or bullets, I think it would be useful to highlight the new features in 5.1 since each seems significant.

**Comment by willingc** [comment]
@minrk When you build these docs locally, are you getting warnings about pygments not finding the ipython lexer? I needed to add ``IPython.sphinxext.ipython_console_highlighting`` explicitly to `extensions` in `conf.py` to remove the warnings.

**Comment by minrk** [comment]
@willingc thanks! I added the highlighting extension and headings per your feedback.

**Comment by willingc** [comment]
@minrk Changes look great. If you haven't cut the release yet, I would like to make one small PR re: the headings symbol hierarchy. Will submit in a few minutes. Worst case, it will go in the next release since it doesn't change display.

**Pull request merged**